### PR TITLE
gap limit + improve comparison of bounds + MOI.RelativeGap

### LINKED ISSUE #417: Set a gap limit in B&P
Hello,

Just a simple question about Coluna optimizer attributes: is it possible to set a gap limit to stop the Branch & Price procedure when the primal and the dual bounds are close enough?

Thank you in advance.
Best,

GD

### PR DESCRIPTION
**Pull request by guimarqu**
Fix #417 

- gap limit of `TreeSearchAlgorithm`
- fix comparisons of bounds : replace `isapprox = |lb - ub| / max(lb, ub) < rtol` by `_gap_closed = |lb - ub| / min(lb, ub) < rtol`
- `MOI.RelativeGap` now available

### CODE REVIEW

File: src/Algorithm/treesearch.jl [hunk 1]
```diff
@@ -23,19 +23,30 @@ getnodevalue(algo::BestDualBoundStrategy, n::Node) = get_ip_dual_bound(getincumb
...
+- `branchingtreefile` : name of the file in which the algorithm writes an overview of the
+branching tree 
 """
 @with_kw struct TreeSearchAlgorithm <: AbstractOptimizationAlgorithm
     conqueralg::AbstractConquerAlgorithm = ColCutGenConquer()
     dividealg::AbstractDivideAlgorithm = SimpleBranching()
     explorestrategy::AbstractTreeExploreStrategy = DepthFirstStrategy()
     maxnumnodes::Int64 = 100000 
     opennodeslimit::Int64 = 100 
+    gap_limit::Float64 = Coluna.DEF_OPTIMALITY_RTOL
```
Comment (rrsadykov) [inline src/Algorithm/treesearch.jl:49]: `gap_limit` is a strange name for me. May be `optimality_tolerance`?

File: src/Algorithm/treesearch.jl [hunk 2]
```diff
@@ -326,7 +337,7 @@ function run!(algo::TreeSearchAlgorithm, rfdata::ReformData, input::Optimization
 
     if treeisempty(tsdata) # it means that the BB tree has been fully explored
         if nb_ip_primal_sols(tsdata.optstate) >= 1
-            if ip_gap_closed(tsdata.optstate) # TODO : add TreeSearch opt tolerances
+            if ip_gap_closed(tsdata.optstate) # TODO : add TreeSearch opt tolerances ?
```
Comment (rrsadykov) [inline src/Algorithm/treesearch.jl:340]: Why no tolerance here?

File: src/Algorithm/treesearch.jl (hunk 1 above)
Comment (guimarqu) [reply src/Algorithm/treesearch.jl:49]: `opt_rtol` and `opt_atol` such as in `ColumnGeneration` ?

File: src/Algorithm/treesearch.jl (hunk 2 above)
Comment (guimarqu) [reply src/Algorithm/treesearch.jl:340]: Ok , I had a doubt but with the new name for `gap_limit`, it makes sense to have tolerance here.

**Review by rrsadykov — APPROVED** [review]

**Pull request merged**